### ISSUE TITLE
Updating title on save?

### ISSUE THREAD

**Issue by bs** [michal-h21/vim-zettel#40]
Thanks for your work on this!

Is there a way to update the %title bit of the filename on save?
(When using let g:zettel_format = "%y%m%d-%H%M-%title")

**Comment by michal-h21** [comment]
Do you mean to update the title when it was changed in the note? Note names should stay static, this is a big difference to normal wiki, where names can be changed. So this is not possible using `vim-zettel` mechanisms. 

Vimwiki has `:VimwikiRenameFile` command. It can rename the file, but I don't think it is that useful for use with Vim-zettel, because you must type the full new name on a prompt. This would be tedious to do with long filenames that contain date and title. I've took a look at the Vimwiki sources to find if it is possible to reuse this functionality, but the code is monolithic and it isn't possible to pass a new filename to the renaming function.

**Comment by irfansharif** [comment]
I wonder if it's worthwhile to have a command that can capture the current file's %title data, and backfill it into the the %title portion of whatever's in zettel_format, using that to then rename the current file. Does vimwiki not provide hooks to propagate the rename to all referencing links?

**Comment by michal-h21** [comment]
Vimwiki provides such command, but it asks for the new name using prompt, it is not possible to pass it as an argument. So you can only copy the new name to a register and use it in `:VimwikiRenameFile` using `Ctrl-r + register name`. 

You can copy a new name using:

     :let @@=zettel#vimwiki#new_zettel_name(zettel#vimwiki#get_title(expand("%")))

You can then enter the new name using `CTRL+r "`.

**Comment by bs** [comment]
Makes sense! It seems that what I often want is my title to be in sync with my top #H1, but perhaps that's not generalized enough.

So :VimwikiRenameFile triggers an update of referencing links. Is this maintained in some index if it happens outside of vim, or only on the method call?

**Comment by michal-h21** [comment]
From reading the Vimwiki source code it seems that it loops over all files in the wiki and does search and replace in each of them.
